R/utils.R

```diff
@@ -45,34 +45,37 @@ getDependency <- function(name, package = name){
   config = sprintf("htmlwidgets/%s.yaml", name)
   jsfile = sprintf("htmlwidgets/%s.js", name)
 
-  config = yaml::yaml.load_file(
-    system.file(config, package = package)
-  )
-  widgetDep <- lapply(config$dependencies, function(l){
-    l$src = system.file(l$src, package = package)
-    do.call(htmlDependency, l)
-  })
+  # if yaml does not exist then assume no dependencies
+  #  in this cases dependencies should be provided through the
+  #  dependencies argument of createWidget
+  widgetDep <- list()
+  if(file.exists(system.file(config, package = package))) {
+    config = yaml::yaml.load_file(
+      system.file(config, package = package)
+    )
+    widgetDep <- lapply(config$dependencies, function(l){
+      l$src = system.file(l$src, package = package)
+      do.call(htmlDependency, l)
+    })
+  }
 
-  bindingDir <- system.file("htmlwidgets", package = package)
-  argsDep <- NULL
-  copyBindingDir <- getOption('htmlwidgets.copybindingdir', TRUE)
-  # TODO: remove this trick when htmltools >= 0.3.3 is on CRAN
-  if (copyBindingDir) {
-    if (packageVersion('htmltools') < '0.3.3') {
-      bindingDir <- tempfile("widgetbinding")
-      dir.create(bindingDir, mode = "0700")
-      file.copy(system.file(jsfile, package = package), bindingDir)
-    } else argsDep <- list(all_files = FALSE)
+  # if js binding does not exist then assume provided through
+  #  some other mechanism such as a specified `htmlDependency` or `script` tag.
+  #  Note, this is a very special case.
+  bindingDep <- NULL
+  if(file.exists(system.file(jsfile, package = package))) {
+    bindingDir <- system.file("htmlwidgets", package = package)
+    argsDep <- NULL
+    bindingDep <- do.call(htmlDependency, c(list(
+      paste0(name, "-binding"), packageVersion(package),
+      bindingDir, script = basename(jsfile)
+    ), argsDep))
   }
-  bindingDep <- do.call(htmlDependency, c(list(
-    paste0(name, "-binding"), packageVersion(package),
-    bindingDir, script = basename(jsfile)
-  ), argsDep))
```

Comment:
The rewrite here is not equivalent to the original code. It threw away `argsDep <- list(all_files = FALSE)`, and the consequence is that everything under the `htmlwidgets/` directory (e.g., https://github.com/rstudio/DT/tree/master/inst/htmlwidgets) will be copyid to `*-binding/` when the widget is rendered, which is not desirable -- only `jsfile` should be copied. Originally this was controlled by the parameter `all_files = FALSE`.

Reply:
BTW, I noticed this issue when I was trying to rebuild the DT website with the latest version of htmlwidgets, and a lot of new files showed up in GIT.